### ISSUE TITLE
Using cargo mutagen in CI pipeline

### ISSUE THREAD

**Issue by dmoka** [llogiq/mutagen#188]
Hey!

I wonder what the options are to use mutagen in CI pipelines. I could not find any support about it such as GitHub actions or so.

The simplest I could do is to run `cargo mutagen` in a CI pipeline task and based on the result I could stop or continue the pipeline run. What would be the easiest way to achieve this, what do you think?

Of course, I am also up to create/submit a PR and change the code regarding, if there is not yet a solution for it.

Thank you!

**Comment by samuelpilz** [comment]
Hi, what are your requirements for `cargo mutagen` for this task?

**Comment by dmoka** [comment]
I meant a similar thing like the 2nd point of this [issue](https://github.com/llogiq/mutagen/issues/156)

Given this said, my posted issue is duplicated as it is a part of the abovementioned issue, so we can close this if you feel so too.

**Comment by samuelpilz** [comment]
Is this something like a `mutationThreshold` which [pit for java](https://pitest.org/quickstart/maven/) has?

**Comment by dmoka** [comment]
What I meant is having different exit codes for the `cargo mutagen` command, so based on that we can see what to do in a CI pipeline run.

This [pit for java](https://pitest.org/quickstart/maven/) seems to be related to the target files, but to be honest I never really used this tool within the java ecosystem....

**Comment by llogiq** [comment]
So `cargo mutagen` should return an error code (instead of success) if at least one mutation survived? Or should we have a threshold?

**Comment by samuelpilz** [comment]
The existence of equivalent mutations makes killing all mutations infeasible. Other mutation testing tools (and also coverage tools) typically use ratio-based thresholds.

**Comment by dmoka** [comment]
@llogiq yes, exactly. If the ratio-based threshold is not reached, then we could return an error code, so the CI pipeline can be stopped.
